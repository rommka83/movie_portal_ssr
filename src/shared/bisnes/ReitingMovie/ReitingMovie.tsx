import React, { FC, HTMLAttributes, PropsWithChildren } from 'react';
import styles from './reitingmovie.module.css';
import classNames from 'classnames';

interface IProps {
  grade: number;
}

export const ReitingMovie: FC<PropsWithChildren<HTMLAttributes<HTMLDivElement> & IProps>> = ({
  grade,
  className,
}) => {
  const [integer, decimal] = grade.toString().split('.');

  return (
    <div className={classNames(styles.root, className)}>
      <span className={styles.left}>{integer},</span>
      <span className={styles.right}>{decimal}</span>
    </div>
  );
};
