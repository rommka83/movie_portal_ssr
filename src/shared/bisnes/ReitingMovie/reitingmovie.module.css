.root {
  display: flex;
  padding-block: 15px;
}
.left {
  font-size: 25px;
  font-weight: 900;
}
.right {
  font-size: 20px;
  font-weight: 900;
}
