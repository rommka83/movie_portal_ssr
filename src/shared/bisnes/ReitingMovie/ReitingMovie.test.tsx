import renderer from 'react-test-renderer';
import { ReitingMovie } from './ReitingMovie';

describe('проверка ReitingMovie', () => {
  test('отрисовка ReitingMovie ', () => {
    const component = renderer.create(<ReitingMovie grade={10} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
