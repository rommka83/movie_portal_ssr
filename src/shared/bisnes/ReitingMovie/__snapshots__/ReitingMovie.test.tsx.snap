// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка ReitingMovie отрисовка ReitingMovie  1`] = `
<div
  className="root"
>
  <span
    className="left"
  >
    10
    ,
  </span>
  <span
    className="right"
  />
</div>
`;
